external/qjsonwrapper/Json.cpp

```diff
@@ -43,7 +43,15 @@ namespace QJsonWrapper
       QMetaProperty metaproperty = metaObject->property( i );
       if ( metaproperty.isReadable() )
       {
-        map[ QLatin1String( metaproperty.name() ) ] = object->property( metaproperty.name() );
+        QVariant val = object->property( metaproperty.name() );
+#if QT_VERSION >= QT_VERSION_CHECK( 6, 0, 0 )
+        if (  ( val.metaType().flags() & QMetaType::IsEnumeration ) &&
+             val.canConvert( QMetaType::Int ) )
```

Comment:
canConvert has a different (stupid) meaning on Qt 6. Now it's "is there any way this variant can possibly be turned into a XXX", and the answer is almost always true. It no longer actually checks the value in the variant to do the test! :scream: 

The qt 6 correct way of doing this is to manually check the userType()

Reply:
(The above did fix things for me, and test_authmethod_authoauth2method)
I'm not totally sure to understand your suggestion. Do you mean just replace the call to metaType() by userType() ? And removing the test ``val.canConvert( QMetaType::Int )`` ? In this generic code, we don't know what exact enumeration type we have, but we want enumeration types to be converted to integers instead of strings (with numeric value) which was what happened without the fix

Reply:
yeah I'd suggest just changing `val.canConvert( QMetaType::Int )` with `val.userType() == QMetaType::Int`

Reply:
> yeah I'd suggest just changing `val.canConvert( QMetaType::Int )` with `val.userType() == QMetaType::Int`

that doesn't work. But I've just removed that test: all enumeration types can be converted to int, right ?